pennylane/measurements/__init__.py

```diff
@@ -11,8 +11,133 @@
+obtained of a given state:
+
+.. code-block:: python
+
+    import pennylane as qml
+    from pennylane.measurements import SampleMeasurement
+
+    class CountState(SampleMeasurement):
+        def __init__(self, state: str):
+            self.state = state  # string identifying the state e.g. "0101"
+            wires = list(range(len(state)))
+            super().__init__(wires=wires)
+
+        def process_samples(self, samples, wire_order, shot_range, bin_size):
+            counts_mp = qml.counts(wires=self._wires)
+            counts = counts_mp.process_samples(samples, wire_order, shot_range, bin_size)
+            return counts.get(self.state, 0)
+
+        def __copy__(self):
+            return CountState(state=self.state)
+
+.. note::
+
+    The ``__copy__`` method needs to be overriden when new arguments are added into the ``__init__``
+    method.
+
+The measurement process in this example uses the :func:`~pennylane.counts` function, which is a
+measurement process which returns a dictionary containing the number of times each quantum
+state has been sampled.
+
+We can now execute the new measurement in a :class:`~pennylane.QNode`. Let's use a simple circuit
+so that we can verify our results mathematically.
+
+.. code-block:: python
+
+    dev = qml.device("default.qubit", wires=1, shots=10000)
+
+    @qml.qnode(dev)
+    def circuit(x):
+        qml.RX(x, wires=0)
+        return CountState(state="1")
+
+.. note::
+
+    In PennyLane we use functions to define measurements (e.g. :func:`counts`). These
+    functions will return an instance of the corresponding measurement process
+    (e.g. :class:`CountsMP`).
+
+The quantum state before the measurement will be:
+
+.. math:: \psi = R_x(\theta) \begin{bmatrix} 1 \\ 0 \end{bmatrix} = \begin{bmatrix}
+                \cos(\theta/2) & -i\sin(\theta/2) \\
+                -i\sin(\theta/2) & \cos(\theta/2)
+            \end{bmatrix} \begin{bmatrix} 1 \\ 0 \end{bmatrix} = \begin{bmatrix}
+                \cos(\theta/2) \\ -i\sin(\theta/2)
+            \end{bmatrix}
+
+When :math:`\theta = 1.23`, the probability of obtaining the state
+:math:`\begin{bmatrix} 1 \\ 0 \end{bmatrix}` is :math:`{\sin(\theta/2)}^2 = 0.333`. Using 10000 shots
+we should obtain the excited state 3333 times approximately.
+
+>>> circuit(1.23)
+tensor(3303., requires_grad=True)
+
+Given that the measurement process returns a real scalar value, we can differentiate it
+using the analytic method.
+
+We know from the previous analysis that the analytic result of the measurement process is
+:math:`r(\theta) = \text{nshots} \cdot \sin^2(\theta/2)`.
+
+The gradient of the measurement process is
+:math:`\frac{\partial r}{\partial \theta} = \text{nshots} \sin(\theta/2) \cos(\theta/2)`.
+
+When :math:`\theta = 1.23`, :math:`\frac{\partial r}{\partial \theta} = 4712.444`
+
+>>> x = qml.numpy.array(1.23, requires_grad=True)
+>>> qml.grad(circuit)(x)
+4715.000000000001
```

Comment:
Alternatively you can put the comment about function and class distinction somewhere at the bottom close to where the actual functions are defined (just below here on the rendered page)

Reply:
I wanted to do this, but couldn't find the way to do it. The functions are rendered automatically, and I didn't find a way to add a comment in this chapter.

Reply:
Done!